control/tests/convert_test.py

```diff
@@ -15,259 +15,238 @@
+        Cmat = ctrb(ssOriginal.A, ssOriginal.B)
+        if (np.linalg.matrix_rank(Cmat) != states):
+            pytest.skip("not reachable")
+        Omat = obsv(ssOriginal.A, ssOriginal.C)
+        if (np.linalg.matrix_rank(Omat) != states):
+            pytest.skip("not observable")
+
+        tfOriginal = tf(ssOriginal)
+        if (verbose):
+            self.printSys(tfOriginal, 2)
+
+        ssTransformed = ss(tfOriginal)
+        if (verbose):
+            self.printSys(ssTransformed, 3)
+
+        tfTransformed = tf(ssTransformed)
+        if (verbose):
+            self.printSys(tfTransformed, 4)
+
+        # Check to see if the state space systems have same dim
+        if (ssOriginal.states != ssTransformed.states) and verbose:
+            print("WARNING: state space dimension mismatch: %d versus %d" %
+                  (ssOriginal.states, ssTransformed.states))
+
+        # Now make sure the frequency responses match
+        # Since bode() only handles SISO, go through each I/O pair
+        # For phase, take sine and cosine to avoid +/- 360 offset
+        for inputNum in range(inputs):
+            for outputNum in range(outputs):
+                if (verbose):
+                    print("Checking input %d, output %d"
+                          % (inputNum, outputNum))
+                ssorig_mag, ssorig_phase, ssorig_omega = \
+                    bode(_mimo2siso(ssOriginal, inputNum, outputNum),
+                         deg=False, plot=False)
+                ssorig_real = ssorig_mag * np.cos(ssorig_phase)
+                ssorig_imag = ssorig_mag * np.sin(ssorig_phase)
+
+                #
+                # Make sure TF has same frequency response
+                #
+                num = tfOriginal.num[outputNum][inputNum]
+                den = tfOriginal.den[outputNum][inputNum]
+                tforig = tf(num, den)
+
+                tforig_mag, tforig_phase, tforig_omega = \
+                    bode(tforig, ssorig_omega,
+                         deg=False, plot=False)
+
+                tforig_real = tforig_mag * np.cos(tforig_phase)
+                tforig_imag = tforig_mag * np.sin(tforig_phase)
+                np.testing.assert_array_almost_equal(
+                    ssorig_real, tforig_real)
+                np.testing.assert_array_almost_equal(
+                    ssorig_imag, tforig_imag)
+
+                #
+                # Make sure xform'd SS has same frequency response
+                #
+                ssxfrm_mag, ssxfrm_phase, ssxfrm_omega = \
+                    bode(_mimo2siso(ssTransformed,
+                                    inputNum, outputNum),
+                         ssorig_omega,
+                         deg=False, plot=False)
+                ssxfrm_real = ssxfrm_mag * np.cos(ssxfrm_phase)
+                ssxfrm_imag = ssxfrm_mag * np.sin(ssxfrm_phase)
+                np.testing.assert_array_almost_equal(
+                    ssorig_real, ssxfrm_real, decimal=5)
```

Comment:
I had to reduce precision here, because running the test with matrix and array advances the random seed and produces different random systems.